shm/src/map.cpp

```diff
@@ -0,0 +1,54 @@
+#include "map.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <random>
+
+constexpr size_t MIN_COORDINATE = 0;
+constexpr size_t MAX_COORDINATE = 10;
+constexpr size_t INITIAL_ISLAND_NUM = 10;
+
+Map::Map() {
+    std::random_device randDevice;
+    std::mt19937 gen(randDevice());
+    std::uniform_int_distribution<> distrib(MIN_COORDINATE,
+                                            MAX_COORDINATE);
+
+    while (vectorOfIslands_.size() != INITIAL_ISLAND_NUM) {
+        size_t posX = distrib(gen);
+        size_t posY = distrib(gen);
+
+        auto it = std::find_if(vectorOfIslands_.begin(),
+                               vectorOfIslands_.end(),
+                               [=](const Island &island) -> bool {
+                                   return (island.getPosition() ==
+                                           Coordinates(posX, posY));
+                               });
+        if (it == vectorOfIslands_.end()) {
+            vectorOfIslands_.push_back(Island(posX, posY));
```

Comment:
It isn't better emplace back?

Reply:
Right, matter of optimisation...
emplace_back() is always as efficient or more efficient than push_back(). It constructs the object in the containers storage directly if possible, as opposed to push_back() which either copies or moves the object.

Reply:
I will use here `emplace_back(poxX, posY)` You will construct directly object in memory. And avoid temporary object.